File: Fahrenheit/src/com/zakharkin/example/lollipop/DBViewer.java
package com.zakharkin.example.lollipop;

import android.app.Activity;
import android.database.Cursor;
import android.database.DataSetObserver;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.GridView;
import android.widget.ListAdapter;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by andre on 22.11.2015.
 */
public class DBViewer extends Activity {

    LogsDataHelper dataHelper;
    GridView gridView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dbviewer);
        dataHelper = new LogsDataHelper(getBaseContext());

        gridView = (GridView) findViewById(R.id.gridView);
        List<String> data = new ArrayList<>();
        ListAdapter la = new ArrayAdapter<>(getBaseContext(), android.R.layout.simple_list_item_1, data);
        SQLiteDatabase database = null;
        try {
            database = dataHelper.getReadableDatabase();
            Cursor c = dataHelper.GetData(database);
            //Move the cursor to the first row.
            if (c.moveToFirst()) {
                do {
                    String dStr = dataHelper.GetString(c);
                    //add in to array list
                    data.add(dStr);
                    gridView.setAdapter(la);
                } while (c.moveToNext());//Move the cursor to the next row.
            } else {
                Log.w("DBViewer", "No data found");
                Toast.makeText(getApplicationContext(), "No data found", Toast.LENGTH_LONG).show();
            }
        } catch (Exception e) {
            Log.e("DBViewer", "No data found" + e.getMessage());
            Toast.makeText(getApplicationContext(), "No data found" + e.getMessage(), Toast.LENGTH_LONG).show();
        } finally {
            if (database != null) {
                database.close();
            }
        }
    }

    public void deleteAllClick(View view){
        dataHelper.deleteAll();
    }
}

File: Fahrenheit/src/com/zakharkin/example/lollipop/DBViewer.kt
package com.zakharkin.example.lollipop

import android.app.Activity
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.*

import java.util.ArrayList

/**
 * Created by andre on 22.11.2015.
 */
class DBViewer : Activity() {

    internal var dataHelper: LogsDataHelper? = null
    internal var gridView: GridView? = null

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dbviewer)
        dataHelper = LogsDataHelper(baseContext)

        gridView = findViewById(R.id.gridView) as GridView

        val la: ListAdapter
        var database: SQLiteDatabase? = null
        try {
            database = (dataHelper as LogsDataHelper).readableDatabase
            val c = (dataHelper as LogsDataHelper).GetData(database)
            la = SimpleCursorAdapter(baseContext, android.R.layout.simple_list_item_2, c, arrayOf<String>("created_at", "logtext"), intArrayOf(android.R.id.text1, android.R.id.text2), 0)
            (gridView as GridView).adapter = la
        } catch (e: Exception) {
            Log.e("DBViewer", "No data found" + e.message)
            Toast.makeText(applicationContext, "No data found" + e.message, Toast.LENGTH_LONG).show()
        } finally {
            if (database != null) {
                database.close()
            }
        }
    }

    fun deleteAllClick(view: View) {
        dataHelper!!.deleteAll()
    }
}